Include/cpython/code.h

```diff
@@ -88,6 +88,7 @@ typedef uint16_t _Py_CODEUNIT;
     PyObject *co_qualname;        /* unicode (qualname, for reference) */      \
     PyObject *co_linetable;       /* bytes object that holds location info */  \
     PyObject *co_weakreflist;     /* to support weakrefs to code objects */    \
+    void *_co_code;               /* cached co_code object/attribute */        \
```

Comment:
Why not `PyObject *`?

Reply:
Same reason as why `co_extra` is `void *`:

```
/*
       Type is a void* to keep the format private in codeobject.c to force     \
       people to go through the proper APIs. */
```
I also didn't want people thinking they can just replace their current `co_code` accesses with `_co_code`. So this should force them to use `PyCode_GetCode` or `PyObject_GetAttrString(co, "co_code")`

Reply:
> For example, we [used `PyObject *` for lazily-computed code attributes](https://github.com/python/cpython/blob/v3.11.0a6/Include/cpython/code.h#L93-L96) during the 3.11 alpha phase (these caches were later removed, though).

Actually, could the removal of the `co_varnames`, `co_cellvars`, and `co_freevars` members be part of the coverage slowdown? I don't know enough about it to know whether they're heavily accessed (though it looks like `co_varnames` is at least sometimes used for "dynamic contexts").

Reply:
This should be a `PyObject *`. No reason to undermine the C compiler's (limited) ability to type check the code.